design-documents/storefront/configuraiton-propagation.md

```diff
@@ -0,0 +1,149 @@
+### 1. Configuration that impacts Admin Panel behavior
+
+Examples: admin ACLs that allow or deny parts of functionality for the admin user, reindex on update or by schedule.
+
+This type of configuration has nothing to do with Storefront and should not be propagated to storefront, as well as should not be exposed via export API. 
+
+### 2. Configuration that impacts data provided by Storefront
+
+Example: Base Media URL is used for calculation of media image URLs returned by Storefront API.
+
+This type of configuration should be propagated to Storefront in one of the following ways:
+
+1. Pre-calculate final data on back office side and provide final result to the Storefront during synchronization. Minimize synchronization by updating only entities that have been really affected.
+   1. Pros: simpler implementation of Storefront due to eliminated necessity for Storefront to keep knowledge about additional configuration, including data calculation algorithms.
+   2. Cons: massive (up to full) reindexation necessary in case the configuration is changed.
+2. Move/duplicate calculation logic in the Storefront service based on original data is synced from the back office.
+    1. Pros and cons are opposite to option 1. So this option is valuable in case of expected frequent change of configuration that impacts a lot of data.
+
+To choose the right approach in a specific case, consider the following:
+
+1. How frequently the configuration is expected to change?
+   1. Frequent configuration changes leading every time to massive reindexation may be unacceptable.
+   2. Configuration changes expected a few times in the store lifespan may not be worth additional complexity on the Storefront side and full reindexation may be better in this case.
+2. Is it acceptable to have significant delay in data propagation after the configuration change?
+   1. Changing Base URL may be not a big issue, especially if a redirect can be setup. So it may be acceptable to have URLs to be fully updated in a few hours.
+   2. Changes in prices, on the other side, may not stand long delays.
+3. Do 3rd-party systems provide similar configuration?
+   1. If 3rd-party systems don't have equivalent configuration, how will it be populated in the Storefront service? It might be better to avoid Magento-specific concepts to simplify integrations, and instead provide indexed data to the Storefront service.
+   1. If a configuration option is pretty common among 3rd-party systems, it may make sense to reflect it in the Storefront service.
+
+Expected consequences are described in the decision document for each case.
+For example, time for configuration propagation in case reindexation is chosen, logic duplication/complexity in case configuration is propagated to Storefront application, performance impact for Storefront read API or for synchronization.
+
+### 3. Configuration that impacts UI representation of Storefront data
+
+Example: include or exclude taxes in displayed prices. 
```

Comment:
After more discussions: taxes incl/excl is not a good example of UI Config, as it's based on the store (which is based on the country it is run in). So It should be part of service data itself - whether taxes are included or excluded.